### ISSUE TITLE
Inconsistent return value when two imports differ only by case

### ISSUE THREAD

**Issue by vesauimonen** [timothycrosley/isort#223]
The isort check passes inconsistently if:
- `order_by_type` config is disabled
- two imports from the same package differ only by their case

Consider the following example:

**.isort.cfg:**
```
[settings]
order_by_type=false
```

**my_file.py:**
```python
from sqlalchemy.dialects.postgresql import ARRAY, array
...
```

**Shell script for running isort:**
```bash
#!/bin/bash

for i in 1 2 3 4 5
do
   echo "isort check no. $i:"
   isort --check-only my_file.py
done
```
returns
```
isort check no. 1:
ERROR: /Users/vesauimonen/Documents/isort-bug/my_file.py Imports are incorrectly sorted.
isort check no. 2:
isort check no. 3:
ERROR: /Users/vesauimonen/Documents/isort-bug/my_file.py Imports are incorrectly sorted.
isort check no. 4:
ERROR: /Users/vesauimonen/Documents/isort-bug/my_file.py Imports are incorrectly sorted.
isort check no. 5:
```

I'm using isort version 3.9.4.

**Comment by timothycrosley** [comment]
Hi @vesauimonen,

I'm unable to reproduce this issue after running tests over 100 times. I went ahead and added a test case to the suite to try and reproduce it - but so far it consistently works. If you can provide more details / a failing test let me know and I will reopen this issue.

Thanks!

~Timothy

**Comment by timothycrosley** [comment]
Hi @vesauimonen,

After adding the test to the test_suite and merging in I can see the issue reproduced within travis occasionally will reopen and think of a fix

Thanks!

~Timothy

**Comment by timothycrosley** [comment]
This has now been fixed in develop:

https://github.com/timothycrosley/isort/pull/245/files

And will be included in the next release.

Thanks!

~Timothy

**Comment by timothycrosley** [comment]
This is now officially fixed in 3.9.6

Thanks again!

~Timothy
